messagebus/src/main/java/com/yahoo/messagebus/network/rpc/RPCTargetPool.java

```diff
@@ -118,14 +122,51 @@ public synchronized int size() {
      * Implements a helper class holds the necessary reference and timestamp of a target. The lastUse member is updated
      * when a call to {@link RPCTargetPool#flushTargets(boolean)} iterates over an active target.
      */
-    private static class Entry {
+    private static class Entry implements Closeable {
 
-        final RPCTarget target;
-        long lastUse = 0;
+        private final RPCTarget [] targets;
+        private int index;
+        long lastUse;
 
-        Entry(RPCTarget target, long lastUse) {
-            this.target = target;
+        Entry(RPCTarget [] targets, long lastUse) {
+            this.targets = targets;
             this.lastUse = lastUse;
         }
+        RPCTarget getTarget(long now) {
+            if (index >= targets.length) {
+                index = 0;
+            }
+            RPCTarget target = targets[index];
```

Comment:
Pure round-robin is presumably a good enough choice here since we're assuming that load to the same target is pretty uniformly distributed across individual requests?

Reply:
That is the assumption, yes.